provider/src/main/kotlin/org/gradle/kotlin/dsl/support/Exceptions.kt

```diff
@@ -22,7 +22,7 @@ import kotlin.reflect.KClass
 
 
 fun illegalElementType(container: NamedDomainObjectCollection<*>, name: String, expectedType: KClass<*>, actualType: KClass<*>) =
```

Comment:
Why not make an exception called `IllegalElementType` which extends `IllegalArgumentException`? Wouldn't that be more descriptive to the user?